cli/cmd/testdata/inject_emojivoto_deployment_injectDisabled.report.verbose

```diff
@@ -1,9 +1,3 @@
+Error transforming resources: Resource with name "web" of kind "deployment" cannot be injected due to the following reasons:
 
-√ pods do not use host networking
-√ pods do not have a 3rd party proxy or initContainer already injected
-‼ "linkerd.io/inject: disabled" annotation set on deployment/web
-‼ no supported objects found
-√ pod specs do not include UDP ports
-
-deployment "web" skipped
-
+× 
```

Comment:
I also happened to notice a very strange bug here. On debugging, I observed that the `Report` object that is returned by the method `newReport()` does hold a string value `"injection_enable_annotation_absent"` under the field `report.InjectDisabledReason`. However, when this report object is returned as a part of `ParseMetaAndYAML()` method and assigned to the variable `report` (in particular, we are looking at [this](https://github.com/linkerd/linkerd2/blob/be558b68694f6e64679562052869d3f7ddd68261/cli/cmd/inject.go#L154) line), `report.InjectDisabledReason` now holds an empty string.
Hence, the method `report.GetInjectFailReason()` also renders an empty string as a part of the error message due to the direct dependency on that particular field.
I'm not really sure if this behaviour would be expected, but it does look like `disableByAnnotation()` method does account for this particular test case,

Reply:
This is pretty weird. Debugging the workflow, showed me that `transform` is being called by twice for the same yaml object once as with `origin` as `CLI` and in other-case with a different value. The `disabledByAnnotation()` of the differentValue is returning as `false`, causing it to not print the reason.

The question is why is transform for a single yaml object twice? Will try to do more debugging on this early tomorrow.

Reply:
Okay, Once it was `uninject` which is called before `inject` but uses the same `processYaml` functions with a different `resourceTransformer`

Reply:
So @Pothulapati @mayankshah1607,  inject generally is (uninject + inject) under the hood. The reason being is that you can actually do a manual inject on an already injected resource only to change something about its injected config. Does that make sense?